src/main/java/com/epam/test_generator/services/LoginService.java

```diff
@@ -5,26 +5,41 @@
 import com.auth0.jwt.JWTVerifier;
 import com.auth0.jwt.algorithms.Algorithm;
 import com.auth0.jwt.interfaces.DecodedJWT;
-import com.epam.test_generator.dao.interfaces.UserDAO;
 import com.epam.test_generator.dto.LoginUserDTO;
 import com.epam.test_generator.entities.User;
+import com.epam.test_generator.services.exceptions.TokenMalformedException;
 import com.epam.test_generator.services.exceptions.UnauthorizedException;
+import java.io.UnsupportedEncodingException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import javax.annotation.Resource;
+import javax.servlet.http.HttpServletRequest;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.PropertySource;
 import org.springframework.core.env.Environment;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
-import javax.annotation.Resource;
-import javax.servlet.http.HttpServletRequest;
-import java.io.IOException;
-import java.io.UnsupportedEncodingException;
-
 @Service
 @PropertySource("classpath:application.properties")
 @Transactional(noRollbackFor = UnauthorizedException.class)
 public class LoginService {
 
+    private static final long REFRESH_TOKEN_DURATION_IN_HOURS = 1L;
+    private static final long EXPIRING_TOKEN_DURATION_IN_HOURS = 24L;
+
+    private static final String ELEMENT_FOR_UNIQUE_TOKEN = "cucumber";
+
+    private static final String CLAIM_ID = "id";
+    private static final String CLAIM_EMAIL = "email";
+    private static final String CLAIM_GIVEN_NAME = "given_name";
+    private static final String CLAIM_FAMILY_NAME = "family_name";
+    private static final String CLAIM_ROLE = "role";
+    private static final String CLAIM_REFRESH_AFTER = "refresh";
```

Comment:
value refresh_at would be better because it explains what the claim is for.

Examples of private claims https://www.iana.org/assignments/jwt/jwt.xhtml